### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by peterschutt**
- Tidy up in `AbstractBaseRepository`
- Removed need to have means for async and sync sqlalchemy connections - removed psycopg2 as dep.
- Isolates unit tests from database.

### CODE REVIEW

File: scripts/entry [hunk 1]
```diff
@@ -33,15 +33,17 @@ postgres_ready() {
   python <<END
```
Comment (Goldziher) [inline scripts/entry:33]: Just a note: regarding running python as a string - i am always dubious about this kind of approach (i know it's not you). One could simply execute a python script in the shell - and then there are all the benefits of code analysis, linting etc. involved. It's also possible to use it in a docker command etc. or call it from a `.sh` entrypoint file.

File: src/app/repositories/base.py [hunk 2]
```diff
@@ -41,13 +89,10 @@ async def get_many(self, *, offset: int = 0, limit: int = 100) -> list[ReturnTyp
...
         """
-        try:
-            results = await self.session.execute(
-                select(self.db_model).offset(offset).limit(limit)
-            )
-        except SQLAlchemyError as e:
-            raise RepositoryException("An exception occurred: " + repr(e)) from e
-        return [self.return_model.from_orm(inst) for inst in results.scalars()]
+        db_models = await self._scalars(
+            select(self.db_model).offset(offset).limit(limit)
```
Comment (Goldziher) [inline src/app/repositories/base.py:93]: very neat

File: scripts/entry (hunk 1 above)
Comment (peterschutt) [reply scripts/entry:33]: Agreed. I had the thought on the way through of defining them in `utils` or somewhere and exposing them as poetry scripts. I'll do that.

File: src/app/config.py [hunk 3]
```diff
@@ -21,51 +22,40 @@ class Config:
...
 
         Returns
         -------
-        PostgresDsn
+        URL
         """
-        return PostgresDsn.build(  # type:ignore[no-any-return]
-            scheme="postgresql",
-            user=self.USER,
+        return URL.create(
```
[the remark this answers is not in the document]
Comment (peterschutt) [reply src/app/config.py:52]: We do the same.. I wasn't sure what others do so this was mostly inertia. Agree with the beginner focus, so I'll leave it.

File: src/app/repositories/base.py [hunk 4]
```diff
@@ -15,6 +18,17 @@
 ReturnType = TypeVar("ReturnType", bound=BaseModel)
 
 
+def wrap_sqla_exception(f: Any) -> Any:
```
[the remark this answers is not in the document]
Comment (peterschutt) [reply src/app/repositories/base.py:21]: Yep good point!

File: src/app/repositories/base.py (hunk 2 above)
Comment (peterschutt) [reply src/app/repositories/base.py:93]: This was mostly just to facilitate easier testing, but I like the way it came out too.

File: src/app/repositories/base.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (peterschutt) [reply src/app/repositories/base.py:21]: Regarding the 2nd point, I'd agree 100% if these methods were longer than a few lines.. given that they are all fairly short, I don't mind it. This was really a bit of experimentation so if you don't feel too  strongly about it I'll leave the decorator in place for now and see how I grow to feel about it.

**Pull request merged**